src/main/kotlin/pl/edu/agh/gem/internal/service/AuthService.kt

```diff
@@ -93,6 +99,25 @@ class AuthService(
     private fun canSendEmail(notVerifiedUser: NotVerifiedUser) =
         notVerifiedUser.codeUpdatedAt.isBefore(now().minus(emailProperties.timeBetweenEmails))
 
+    @Transactional
+    fun sendPasswordRecoveryEmail(email: String) {
+        val verifiedUser = verifiedUserRepository.findByEmail(email) ?: throw UserNotFoundException()
+
+        if (passwordRecoveryCodeRepository.findByUserId(verifiedUser.id) != null) {
+            throw EmailRecentlySentException()
+        }
+
+        val passwordRecoveryCode = passwordRecoveryCodeRepository.create(
+            PasswordRecoveryCode(
+                userId = verifiedUser.id,
+                code = generateCode(),
+            ),
+        )
+
+        val link = "http://${urlProperties.gemUrl}/$OPEN/send-password?email=$email&code=${passwordRecoveryCode.code}"
```

Comment:
could you move that link to the client (not domain part to know where to know that)

Reply:
done